### DISCUSSION TITLE
Use this as primary mongodb for the application

### DISCUSSION THREAD

**Discussion by pranavburnwal** [typegoose/mongodb-memory-server#884] [General]
Read: https://blog.screenshotbot.io/2024/08/10/building-a-highly-available-web-service-without-a-database/

I was thinking of building in memory Mongo base (maybe with Mongoose).
Mongo already has a 100% in-memory option as stated here: https://www.mongodb.com/docs/manual/core/inmemory/

Was thinking of building a replica set of maybe 3 (or more)
As defined in: https://www.mongodb.com/docs/manual/core/inmemory/#replica-set

What are your thoughts?

**Comment by hasezoey** [comment]
I dont quite know what answer you are expecting, but let me clarify that despite the name `mongodb-memory-server`, this library is a downloader & manager of mongodb binaries, which for a long time made use of `ephermeralForTest` which was a free*1 in-memory engine in mongodb, but it got removed with 7.0.

*1 free as in available in the Community edition.

Additionally, the [`inMemory` engine](https://www.mongodb.com/docs/manual/core/storage-engines/) you linked is for mongodb *Enterprise* only, this library here manages binaries of the *Community* edition and has not been tested with the Enterprise version.

> An [In-Memory storage engine](https://www.mongodb.com/docs/manual/core/inmemory/#std-label-storage-inmemory) is available in MongoDB Enterprise. Rather than storing documents on-disk, it retains them in-memory for more predictable data latencies.

PS: i only have skimmed the linked blogpost

**Reply by pranavburnwal** [reply]
Gotcha @hasezoey, thanks for sharing! 
Will try to get PoC up for this and see if I can make it work.
